pybliometrics/scopus/scopus_search.py

```diff
@@ -215,6 +217,179 @@ def get_eids(self):
+                if bib_tex_type == None:
+                    raise ValueError(f"Unsupported type | Document type: {document_type} | Aggregation type: {aggregation_type} | DOI: https://doi.org/{result.doi}")
+
+
+                fields = []
+
+                fields = self.add_bibtex_field(fields, "author", authors)
+                fields = self.add_bibtex_field(fields, "title", result.title)
+                fields = self.add_bibtex_field(fields, "date", result.coverDate)
+                if aggregation_type == aggregation_type_journal:
+                    fields = self.add_bibtex_field(fields, "journal", result.publicationName)
+                    fields = self.add_bibtex_field(fields, "volume", result.volume)
+                    fields = self.add_bibtex_field(fields, "number", result.issueIdentifier)
+                elif aggregation_type == aggregation_type_conference_proceedings or aggregation_type == aggregation_type_book_series:
+                    fields = self.add_bibtex_field(fields, "booktitle", result.publicationName)
+                elif bib_tex_type == bib_tex_type_techreport:
+                    fields = self.add_bibtex_field(fields, "institution", affiliation)
+                fields = self.add_bibtex_field(fields, "pages", pages)
+                fields = self.add_bibtex_field(fields, "doi", result.doi)
+                fields = self.add_bibtex_field(fields, "url", "https://api.elsevier.com/content/abstract/scopus_id/" + result.eid.rsplit("-", 1)[1])
+                if not bib_tex_type == bib_tex_type_techreport:
+                    fields = self.add_bibtex_field(fields, "affiliation", affiliation)
+                fields = self.add_bibtex_field(fields, "abstract", result.description)
+                fields = self.add_bibtex_field(fields, "author_keywords", author_keywords)
+                if bib_tex_type == bib_tex_type_book:
+                    fields = self.add_bibtex_field(fields, "isbn", result.volume)
+                fields = self.add_bibtex_field(fields, "issn", result.issn)
+                fields = self.add_bibtex_field(fields, "type", document_type)
+                fields = self.add_bibtex_field(fields, "scopus_aggregation_type", aggregation_type)
+                fields = self.add_bibtex_field(fields, "citedby_count", result.citedby_count)
+                fields = self.add_bibtex_field(fields, "openaccess", result.openaccess)
+                fields = self.add_bibtex_field(fields, "fund_sponsor", result.fund_sponsor)
+                fields = self.add_bibtex_field(fields, "source", "Scopus")
+
+                entry = bibtexparser.model.Entry(bib_tex_type, key, fields)
+
+                bib_library.add(entry)
+
+                # Check whether the addition was successful or resulted in a duplicate block that needs fixing.
+                for i in range(26):
+                    failed_blocks = bib_library.failed_blocks
+                    if failed_blocks:
```

Comment:
What exactly is happening here?

Reply:
The used python library has problems adding a new entry if the key already exists in the library. So if there is already "@Author" in the library, no other entry with "@Author" can be added. Then, the addition will result in a failed block. This checks whether the addition failed. If so, we try to add the entry as "@Authora" instead. If that fails, it will just move through the alphabet until it reaches "@Authorz". Not ideally, but this is mainly dealing with shortcomings of the used library...